### PR TITLE
feat: add ResponsiveIconButton component

### PR DESCRIPTION
**Pull request by HugoPerard**
Adding a new component : ResponsiveIconButton
This component simplify developer experience for integrate button in responsive interface, you can define custom breakpoint for decides when Button component is replace by an IconButton component

Capture d’écran 2022-02-25 à 15 20 45

### CODE REVIEW

File: src/components/ResponsiveIconButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { Button, ButtonProps, IconButton } from '@chakra-ui/button';
+import { useBreakpointValue } from '@chakra-ui/media-query';
+
+interface BreakpointsProps {
+  base?: boolean;
+  sm?: boolean;
+  md?: boolean;
+  lg?: boolean;
+  xl?: boolean;
+}
```
Comment (yoannfleurydev) [inline src/components/ResponsiveIconButton/index.tsx:10]: Isn't that already available through Chakra UI typings ?

File: src/components/ResponsiveIconButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+interface ResponsiveIconButtonProps extends ButtonProps {
+  hideTextBreakpoints?: BreakpointsProps;
+  icon: React.ReactElement;
+  children?: string;
+  iconPosition?: 'left' | 'right';
+  to?: string;
```
Comment (yoannfleurydev) [inline src/components/ResponsiveIconButton/index.tsx:16]: Looks like it is not used.

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

File: src/components/ResponsiveIconButton/docs.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import { Stack } from '@chakra-ui/react';
```
Comment (yoannfleurydev) [inline src/components/ResponsiveIconButton/docs.stories.tsx:3]: Not used

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

**Comment by yoannfleurydev** [comment]
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=178&id=BearStudio_start-ui&open=AX8xQitHt0XDxvfrvkTT

File: src/components/index.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ export * from './Icons';
 export * from './InputCurrency';
 export * from './Logo';
 export * from './Nav';
+export * from './ResponsiveIconButton';
 export * from './Pagination';
```
Comment (yoannfleurydev) [inline src/components/index.ts:23]:
```suggestion
export * from './Pagination';
export * from './ResponsiveIconButton';
```

**Review by yoannfleurydev — APPROVED** [review]

**Review by ivan-dalmet — APPROVED** [review]

**Pull request merged**